src/detect-content.c

```diff
@@ -506,36 +506,62 @@ void DetectContentPropagateLimits(Signature *s)
 
                     if (cd->flags & DETECT_CONTENT_DISTANCE && cd->distance >= 0) {
                         if ((uint32_t)offset_plus_pat + cd->distance <= UINT16_MAX) {
-                            offset = cd->offset = offset_plus_pat + cd->distance;
+                            offset = cd->offset = (uint16_t)(offset_plus_pat + cd->distance);
                         } else {
                             SCLogDebug("not updated content offset as it would overflow : %u + %d", offset_plus_pat, cd->distance);
                         }
                         SCLogDebug("updated content to have offset %u", cd->offset);
                     }
                     if (have_anchor && !last_reset && offset_plus_pat && cd->flags & DETECT_CONTENT_WITHIN && cd->within >= 0) {
                         if (depth && depth > offset_plus_pat) {
-                            uint16_t dist = 0;
+                            int32_t dist = 0;
                             if (cd->flags & DETECT_CONTENT_DISTANCE && cd->distance > 0) {
                                 dist = cd->distance;
                                 SCLogDebug("distance to add: %u. depth + dist %u", dist, depth + dist);
                             }
                             SCLogDebug("depth %u + cd->within %u", depth, cd->within);
-                            depth = cd->depth = depth + cd->within + dist;
+                            if (depth + cd->within + dist < 0 ||
```

Comment:
I would like these to go into their own commit

Reply:
ok, looks relevant